pkg/plugins/policies/meshhttproute/plugin/v1alpha1/plugin.go

```diff
@@ -119,6 +123,9 @@ func ApplyToGateway(
 	rules []ToRouteRule,
 ) error {
 	var limits []plugin_gateway.RuntimeResoureLimitListener
+	if len(rules) == 0 {
+		return nil
+	}
 
 	for _, info := range plugin_gateway.ExtractGatewayListeners(proxy) {
```

Comment:
This function `ExtractGatewayListeners` reuses some old part of gateway code which has services filtered by the traffic permissions